src/threading.c

```diff
@@ -1008,6 +1008,40 @@ JL_DLLEXPORT int jl_heartbeat_enable(int heartbeat_s, int show_tasks_after_n,
     return 0;
 }
 
+// temporarily pause the heartbeat thread
+JL_DLLEXPORT int jl_heartbeat_pause(void)
+{
+    if (!heartbeat_enabled) {
+        return -1;
+    }
+    heartbeat_enabled = 0;
+    return 0;
+}
+
+// resume the paused heartbeat thread
+JL_DLLEXPORT int jl_heartbeat_resume(void)
+{
+    // cannot resume if the heartbeat thread is already running
+    if (heartbeat_enabled) {
+        return -1;
+    }
+
+    // cannot resume if we weren't paused (disabled != paused)
+    if (heartbeat_interval_s == 0) {
+        return -1;
+    }
+
+    // heartbeat thread must be ready
+    if (uv_sem_trywait(&heartbeat_off_sem) != 0) {
+        return -1;
+    }
+    n_hbs_missed = 0;
+    n_hbs_recvd = 0;
```

Comment:
is it worth a comment saying why resetting these is the right thing to do? I guess if we're resuming then we've been paused for some period of time and consider the "heartbeats lost/recovered" metrics to be voided... i wonder if it's a bit easier to follow if we do this inside `pause`, or maybe a comment is enough?

Also why don't we need to reset `tasks_showed`?

Reply:
Added a comment. You're right that it would be perfectly valid to do this in `pause`; don't think it matters though.

Added reset of `tasks_showed` also.